src/components/SingleTodo.tsx

```diff
@@ -1,77 +1,111 @@
-import React, { useEffect, useRef, useState } from "react";
-import { Todo } from "../model";
-import { AiFillEdit, AiFillDelete } from "react-icons/ai";
-import { MdDone } from "react-icons/md";
-
-type Props = {
-  todo: Todo;
-  inputs: Todo[];
-  setInputs: React.Dispatch<React.SetStateAction<Todo[]>>;
-};
-const SingleTodo = ({ todo, inputs, setInputs }: Props) => {
-  const [edit, setEdit] = useState<boolean>(false);
-  const [editTodo, setEditTodo] = useState<string>(todo.todo);
-  const inputRef = useRef<HTMLInputElement>(null);
-  useEffect(() => {
-    inputRef.current?.focus();
-  }, [edit]);
-  const Edit = (e: React.FormEvent, id: number) => {
-    e.preventDefault();
-    setInputs(
-      inputs.map((todo) =>
-        todo.id === id ? { ...todo, todo: editTodo } : todo
-      )
-    );
-    setEdit(false);
-  };
-  const Delete = (id: number) => {
-    setInputs(inputs.filter((todo) => todo.id !== id));
-  };
-
-  const Done = (id: number) => {
-    setInputs(
-      inputs.map((todo) =>
-        todo.id === id ? { ...todo, isDone: !todo.isDone } : todo
-      )
-    );
-  };
-
-  return (
-    <form className="todos--single" onSubmit={(e) => Edit(e, todo.id)}>
-      {edit && ! todo.isDone && (
-        <input
-        value={editTodo}
-        onChange={(e) => setEditTodo(e.target.value)}
-        className="todos--single--text"
-        ref={inputRef}
-      />
-      )}
-      {!edit &&  todo.isDone && (
-        <s className="todos--single--text">{todo.todo}</s>
-      )}
-      {!edit && !todo.isDone && (
-        <span className="todos--single--text">{todo.todo}</span>
-      )}
-      <div>
-        <span
-          className="icon"
-          onClick={() => {
-            if (!edit && !todo.isDone) {
-              setEdit(!edit);
-            }
-          }}
-        >
-          <AiFillEdit />
-        </span>
-        <span className="icon" onClick={() => Delete(todo.id)}>
-          <AiFillDelete />
-        </span>
-        <span className="icon" onClick={() => Done(todo.id)}>
-          <MdDone />
-        </span>
-      </div>
-    </form>
-  );
-};
-
-export default SingleTodo;
+import React, { useEffect, useRef, useState } from "react";
+import { Todo } from "../model";
+import { AiFillEdit, AiFillDelete } from "react-icons/ai";
+import { MdDone } from "react-icons/md";
+import styled from "styled-components";
+
+const Single = styled.form`
+  display: flex;
+  border-radius: 5px;
+  padding: 20px;
+  margin-top: 15px;
+  transition: 0.2s;
+
+  &:hover {
+    box-shadow: 0 0 5px black;
+    transform: scale(1.03);
+  }
+`;
+const ISingleText = styled.input`
+  flex: 1;
+  padding: 5px;
+  border: none;
+  font-size: 20px;
+
+  &:focus {
+    outline: none;
+  }
+`;
+const SingleText = styled.span`
+  flex: 1;
+  padding: 5px;
+  border: none;
+  font-size: 20px;
+
+  &:focus {
+    outline: none;
+  }
+`;
+
+const Icon = styled.span`
+  margin-left: 10px;
+  font-size: 25px;
+  cursor: pointer;
+`;
```

Comment:
Try to use more descriptive naming for components.
The Single should be called form, ISingleText -> Input, SingleText -> Text